src/Operations/InjectStrictDynamic.php

```diff
@@ -26,29 +26,31 @@ public function __construct(array $allowedCSPHashAlgs)
     /**
      * Transform the given set of headers
      *
-     * @param HeaderBag $headers
+     * @param HeaderBag $HeaderBag
      * @return void
      */
-    public function modify(HeaderBag &$headers)
+    public function modify(HeaderBag &$HeaderBag)
     {
-        $CSPHeaders = $headers->getByName('content-security-policy');
+        $CSPHeaders = array_merge(
+            $HeaderBag->getByName('content-security-policy'),
+            $HeaderBag->getByName('content-security-policy-report-only')
+        );
 
-        if (isset($CSPHeaders[0]))
+        foreach ($CSPHeaders as $Header)
         {
-            $header = $CSPHeaders[0];
-
-            $directive = $this->canInjectStrictDynamic($header);
+            $directive = $this->canInjectStrictDynamic($Header);
 
             if (is_string($directive))
             {
-                $header->setAttribute($directive, "'strict-dynamic'");
+                $Header->setAttribute($directive, "'strict-dynamic'");
             }
             elseif ($directive !== -1)
             {
                 $this->addError(
-                    "<b>Strict-Mode</b> is enabled, but <b>'strict-dynamic'</b>
-                        could not be added to the Content-Security-Policy
-                        because no hash or nonce was used.",
+                    "<b>Strict-Mode</b> is enabled, but
+                    <b>'strict-dynamic'</b> could not be added to <b>"
+                    . $Header->getFriendlyName()
+                    . '</b> because no hash or nonce was used.',
                     E_USER_WARNING
```

Comment:
Might be something to think about for `3.0`, would be a BC break to start doing it now though.

Things to think about with errors v.s. exceptions:
- Exceptions will generally cause headers not to reach the adapter, and so won't be sent. i.e. 
   - Is no CSP better than suboptimal CSP? (bearing in mind loss of reporting header which might be stricter too).
   - Is not sending a suboptimal CSP better than HSTS, Referrer-Policy, cookie upgrades, ...? 
- It we throw exceptions, they should be genuine problems. That means not throwing an exception for advice: we cannot use exceptions to tell the dev they they should use something (learning opportunity). Learning opportunities would have to be disposed of if we converted all errors as used currently to exceptions.